pkg/restapi/edv/operation/operations_test.go

```diff
@@ -254,15 +263,61 @@ func TestCreateDocumentHandler_InvalidStructuredDocumentJSON(t *testing.T) {
 	require.Contains(t, rr.Body.String(), "EOF")
 }
 
+func TestCreateDocumentHandler_DocIDIsNotBase58Encoded(t *testing.T) {
+	op := New(memstore.NewProvider())
+
+	createDataVaultExpectSuccess(t, op)
+
+	req, err := http.NewRequest("POST", "", bytes.NewBuffer([]byte(testStructuredDocumentWithNonBase58ID)))
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+
+	urlVars := make(map[string]string)
+	urlVars[vaultIDPathVariable] = testVaultID
+
+	req = mux.SetURLVars(req, urlVars)
+
+	createDocumentEndpointHandler := getHandler(t, op, createDocumentEndpoint)
+
+	createDocumentEndpointHandler.Handle().ServeHTTP(rr, req)
+
+	require.Equal(t, http.StatusBadRequest, rr.Code)
+	require.Equal(t, NotBase58EncodedErrMsg, rr.Body.String())
+}
+
+func TestCreateDocumentHandler_DocIDWasNot128BitsBeforeEncodingAsBase58(t *testing.T) {
+	op := New(memstore.NewProvider())
+
+	createDataVaultExpectSuccess(t, op)
+
+	req, err := http.NewRequest("POST", "",
+		bytes.NewBuffer([]byte(testStructuredDocumentWithIDThatWasNot128BitsBeforeBase58Encoding)))
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+
+	urlVars := make(map[string]string)
+	urlVars[vaultIDPathVariable] = testVaultID
+
+	req = mux.SetURLVars(req, urlVars)
+
+	createDocumentEndpointHandler := getHandler(t, op, createDocumentEndpoint)
+
+	createDocumentEndpointHandler.Handle().ServeHTTP(rr, req)
+
+	require.Equal(t, http.StatusBadRequest, rr.Code)
+	require.Equal(t, Not128BitValueErrMsg, rr.Body.String())
+}
+
 func TestCreateDocumentHandler_DuplicateDocuments(t *testing.T) {
 	op := New(memstore.NewProvider())
 
 	createDataVaultExpectSuccess(t, op)
 
 	storeStructuredDocumentExpectSuccess(t, op)
 
-	req, err := http.NewRequest("POST", "/encrypted-data-vaults/"+testVaultID+"/docs",
-		bytes.NewBuffer([]byte(testStructuredDocument)))
+	req, err := http.NewRequest("POST", "", bytes.NewBuffer([]byte(testStructuredDocument)))
```

Comment:
These changes to the http.NewRequest call are just me doing a quick refactor. The "URL" argument isn't actually used since the endpoint is called directly, so I removed it.